### ISSUE TITLE
Dummy Output - Missing tplg: sof-adl-rt711-2ch.tplg - HP Omen 16-k00023dx, Intel i7-12700H, Mint 21.1

### ISSUE THREAD

**Issue by Warrick126** [thesofproject/linux#4135]
Hello,

My sound card does not show up under sound devices, only Dummy Output. The Mint forums pointed me to a few lines in a log:

[    4.855855] sof-audio-pci-intel-tgl 0000:00:1f.3: Direct firmware load for intel/sof-tplg/sof-adl-rt711-2ch.tplg failed with error -2
[    4.855860] sof-audio-pci-intel-tgl 0000:00:1f.3: error: tplg request firmware intel/sof-tplg/sof-adl-rt711-2ch.tplg failed err: -2
[    4.855866] sof-audio-pci-intel-tgl 0000:00:1f.3: you may need to download the firmware from https://github.com/thesofproject/sof-bin/

I looked in the directory and in the download from thesofproject, but was unable to find that file. I see sof-adl-rt711-4ch.tplg, and others similarly named, but not 2ch. I guess that tplg hasn't been created yet?

I've attached the output from dmesg. Please let me know if you need more. I'm a novice when it comes to Linux, but I can use the terminal well enough.
[dmesgOutput.txt](https://github.com/thesofproject/linux/files/10388117/dmesgOutput.txt)

Thank you for any help you can provide.

**Comment by plbossart** [comment]
It's likely the same issue as https://github.com/thesofproject/linux/issues/4088

HP and Intel NUC usually share the same bad config at the BIOS level. I see the same RT1316 and RT711 listed, the topology that's requested is just not the right one.

Can you add this file 
[sof-dyndbg.conf.txt](https://github.com/thesofproject/linux/files/10388241/sof-dyndbg.conf.txt) as /etc/modprobe.d/sof-dyndbg.conf, reboot and attach the full dmesg log. Also attach the result of 'alsa-info' so that we can prepare a quirk.

you will need to compile your own kernel to test the solution. Fear not, there's a documentation to help
https://thesofproject.github.io/latest/getting_started/setup_linux/install_locally.html

**Comment by Warrick126** [comment]
Ah, I saw another issue that mentioned a bad BIOS config and I wondered if it might be related.

[alsa-info.txt](https://github.com/thesofproject/linux/files/10388467/alsa-info.txt)
[dmesgOutput2.txt](https://github.com/thesofproject/linux/files/10388469/dmesgOutput2.txt)

Well, I guess I get to learn another new trick in linux. :)

**Comment by plbossart** [comment]
~~ @Warrick126 can you re-attach the files, I can't see them. Something bad happened in the initial comment and I tried but failed to find the file.~~

Never mind, brain overload here.

**Comment by plbossart** [comment]
Actually we already have a quirk for this
````
	/* ADL devices */
	{
		.matches = {
			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
			DMI_MATCH(DMI_PRODUCT_NAME, "OMEN by HP Gaming Laptop 16-k0xxx"),
		},
		.driver_data = (void *)hp_omen_16,
	},
````

so my guess is that your kernel is too old.

commit df55100551a34bddab02dff48d0296bda0659c02 was added after v6.0-rc1 so should be in 6.1

try to update to 6.1

**Comment by plbossart** [comment]
The initial issue for this device was https://github.com/thesofproject/linux/issues/3727

**Comment by Warrick126** [comment]
Neat, I'll see if I can. I'm currently running 6.0.0-1010

**Comment by Warrick126** [comment]
So far 6.1.0-1004-oem is running fine, but my speakers still didn't work. 

I looked in the log and saw the same missing file error. However, this time it was asking for a different file, sof-adl-rt711-l0-rt1316-l3-2ch.tplg. I searched my download from github and found it there. I then copied it to the /lib/firmware/intel/sof-tplg directory and rebooted. After that, the sound worked! I checked the headphone combo jack too and that worked. So, I think I'm good to go. Thank you for your help!
